example/z-docker-opam/Dockerfile

```diff
@@ -0,0 +1,32 @@
+FROM ocaml/opam:alpine as build
+
+# Install system dependencies
+# Some of these are only needed for database and JavaScript stuff, but this is 
+# our build context, it won't impact our runtime environment.
+RUN sudo apk add --update \
+    make libffi-dev linux-headers m4 ncurses-dev perl pkgconf postgresql-dev \
+    musl-dev git pkgconfig gmp-dev libc-dev openblas-dev zlib-dev \
+    npm nodejs libev-dev
+
+RUN eval $(opam env) && opam repository set-url default https://opam.ocaml.org && opam update
+
+WORKDIR /build
+
+# Install dependencies
+# We only copy the opam file to install dependencies only when they change, and 
+# not every time another file is modified.
+ADD hello.opam hello.opam
+RUN eval $(opam env) && opam install . --deps-only
+
+# Build project
+ADD . .
+
+RUN opam exec -- dune build --profile release
```

Comment:
There is no need for `--profile release` AFAIK &mdash; it just stops treating warnings as errors. If someone wants that, they can add it later, but it's not a big need for leaf projects AFAIK.

Reply:
Remove!